beacon-chain/blockchain/execution_engine.go

```diff
@@ -233,6 +233,8 @@ func (s *Service) notifyNewPayload(ctx context.Context, postStateVersion int,
 // reportInvalidBlock deals with the event that an invalid block was detected by the execution layer
 func (s *Service) reportInvalidBlock(ctx context.Context, root, parentRoot, lvh [32]byte) error {
 	newPayloadInvalidNodeCount.Inc()
+	s.cfg.ForkChoiceStore.Lock()
+	defer s.cfg.ForkChoiceStore.Unlock()
```

Comment:
I am afraid this can be missused, I can see someone adding this `reportInvalidBlock` somewhere where it's already locked. What do you think about just removing `reportInvalidBlock` and have it inside `validateExecutionOnBlock`. I dont think this needs to be another function

Reply:
kept a different function but addded a chain-info helper that locks.